src/syrupy/report.py

```diff
@@ -67,6 +75,43 @@ def __attrs_post_init__(self) -> None:
                 else:
                     self.failed.update(snapshot_fossil)
 
+    def __parse_invocation_args(self) -> None:
+        arg_groups: List[Tuple[Optional[str], str]] = []
+        path_as_package = False
+        maybe_opt_arg = None
+        for arg in self._invocation_args:
+            if arg.strip() == "--pyargs":
+                path_as_package = True
+            elif arg.startswith("-"):
+                if "=" in arg:
+                    arg0, arg1 = arg.split("=")
+                    arg_groups.append((arg0.strip(), arg1.strip()))
+                elif maybe_opt_arg is None:
+                    maybe_opt_arg = arg
+                    continue  # do not reset maybe_opt_arg
+            else:
+                arg_groups.append((maybe_opt_arg, arg.strip()))
+
+            maybe_opt_arg = None
+
+        for maybe_opt_arg, arg_value in arg_groups:
+            if maybe_opt_arg == "-k":  # or maybe_opt_arg == "-m":
```

Comment:
Ignoring the marker expression as they can not be used to match against snapshot names and are already used to select which pytest items in the collection to run.